src/service.rs

```diff
@@ -619,6 +651,24 @@ where
             };
 
             tracing::info!("Got vid commitment for block {id}",);
+
+            // This block was truncated, but we got VID in time with margin left.
+            // Maybe we can handle bigger blocks?
+            if block_info.truncated
+                && timeout_after.duration_since(Instant::now())
+                    > self.max_api_waiting_time / VID_RESPONSE_TARGET_MARGIN_DIVISOR
+            {
+                // Increase max block size
+                let mut global_state = self.global_state.write_arc().await;
+                global_state.max_block_size = std::cmp::min(
+                    global_state.max_block_size
+                        + global_state
+                            .max_block_size
+                            .div_ceil(MAX_BLOCK_SIZE_CHANGE_DIVISOR),
+                    MAX_BLOCK_SIZE_FLOOR,
```

Comment:
Shouldn't this be `MAX_BLOCK_SIZE_CEIL` (not sure if we have such value)?

Reply:
Oh, good catch! There's no reason to make such a ceiling, the whole min call is copypasta error.